crates/sui-adapter/src/adapter.rs

```diff
@@ -77,9 +78,25 @@ pub fn execute<
             CallArg::Pure(_) => None,
             CallArg::Object(ObjectArg::ImmOrOwnedObject((id, _, _)))
             | CallArg::Object(ObjectArg::SharedObject(id)) => {
-                Some((*id, state_view.read_object(id)?))
+                Some(vec![(*id, state_view.read_object(id)?)])
+            }
+            CallArg::ObjVec(vec) => {
+                if vec.is_empty() {
+                    return None;
+                }
```

Comment:
Fine here, but should we ban this in the transaction input checker?

Reply:
It should be OK to pass an empty vector (of internal object type) to an entry function (I even have a test for this) so I am not sure we should ban it.

Reply:
Ah sure, I could see it being useful